static/app/components/searchQueryBuilder/hooks/useQueryBuilderState.tsx

```diff
@@ -862,6 +863,68 @@ export function useQueryBuilderState({
               newState.query !== state.query && displayAskSeerFeedback ? true : false,
           };
         }
+        case 'UPDATE_FREE_TEXT_ON_BLUR': {
+          const newState = updateFreeText(state, action);
+
+          if (
```

Comment:
Can you extract the logic here into functions and add a comment about what is different between them and why it is necessary? Seems helpful to refer back to in the future